src/batou_ext/modsecurity.py

```diff
@@ -0,0 +1,105 @@
+import batou.component
+import batou.lib.file
+import batou_ext.nix
+import os
+
+
+@batou_ext.nix.rebuild
+class ModSecurity(batou.component.Component):
+    '''
+    Manages one or more mod_security rule sets for nginx, including the "OWASP Core Rule Set".
+
+    Without any parameters this creates a basic default rules file on the lowest paranoia
+    level and logs events but does not block anything.
+
+    Specific rule files need to be enabled through your project's nginx configuration,
+    typically by enabling mod_security and choosing the profile:
+
+         server {
+             modsecurity on;
+             modsecurity_rules_file /etc/modsecurity_includes_default.conf
+         }
+
+
+    ## Default usage:
+
+         self += Modsecurity()
+
+    This uses the default behaviour:
+
+    - Modsecurity does not block any request, it just logs
+    - The Paranoia Level is at 1 which is the lowest setting
+
+
+    ## Customize attributes
+
+    You may change these by giving other attribute values.
+
+          self += Modsecurity(block=True, paranoia_level=20)
+
+    ## Adding new rules
+
+    At this Paranoia Level it may be necessary to change some configuration
+    to negate false positives. You may add new configuration files, which will be placed
+    at the bottom of the includes file. This is done with `includes`, which expects a list
+    of strings with the path to the file, or if just one file is added, only a string.
+
+          self += Modsecurity(
+              block=True,
+              paranoia_level=20,
+              includes=['path/to/file1', 'path/to/file2']
+          )
+
+    ## Multiple instances
+
+    It is completly possible to use multiple different rulesets on the same server.
+    This is done with the `ruleset_name` attribute. If no value is given it uses `default`.
+
+          self.modsecurity_frontend_1 = Modsecurity(
+              ruleset_name='frontend1',
+              paranoia_level=20,
+              includes='frontend1_exceptions'
+          )
+
+          self.modsecurity_frontend_2 = Modsecurity(
+              ruleset_name='frontend2',
+              block=True,
+              includes=['frontend2_exceptions', 'frontend2_additions']
+          )
+
+    You can get he includes path for each instance of ModSecurity
+    from its `include_path` attribute.
+    '''
+
+    block = False
+    paranoia_level = 1
+    ruleset_name = 'default'
+    include_path = f'/etc/modsecurity_includes_{ruleset_name}.conf'
+    enable_owasp = True
+    # current on branch v3.3/master
+    owasp_revision = '18703f1bc47e9c4ec4096853d5fb4e2a204a07a2'
+    includes = None
+
+    def configure(self):
+
+        if isinstance(self.includes, str):
```

Comment:
remove the include mechanism for now as you haven't used it anywhere yet. 

what would be good: actually go and get a specific use case from a customer installation where we need to respond to increased paranoia levels. I agree that it will be annoying to quickly implement this when its actually needed, but without proof that this is what is helpful in that scenario, I don't buy that it's the right way. It could be, but it doesn't have to be and I hate dead code.